logstash-core/src/main/java/org/logstash/config/ir/compiler/EventCondition.java

```diff
@@ -331,6 +328,12 @@ private static EventCondition eq(final EventValueExpression evalE,
                 return new EventCondition.Compiler.FieldEqualsLong(
                     field, ((Number) value).longValue()
                 );
+            } else if (value instanceof List) {
+                if (!((List) value).isEmpty()) {
+                    // shouldn't get here; this is pre-validated in the lscl.
+                    throw new IllegalStateException("non-empty arrays are not allowed in boolean expressions");
```

Comment:
Thought: Maybe we should explain to the user why? I'd be confused and maybe upset at this seemingly-arbitrary restriction.

Reply:
yeah; makes sense.

```
non-empty arrays are not allowed in boolean expressions because the Logstash syntax in this context cannot unambiguously differentiate between a field selector and a single-element array.
```

More wordy, but indicates that we're not just being jerks :/